src/Mvc/Mvc.Core/src/Infrastructure/ProblemDetailsApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Core;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Extension methods for <see cref="IApplicationBuilder"/> to add support for sending ProblemDetails
+    /// payloads back for 500 and 404 errors.
+    /// </summary>
+    public static class ProblemDetailsBuilderExtensions
+    {
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="app"></param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseProblemDetails(this IApplicationBuilder app)
+        {
+            if (app == null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            return app.UseMiddleware<ProblemDetailsMiddleware>();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="app"></param>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseProblemDetails(
+            this IApplicationBuilder app,
+            ProblemDetailsOptions options)
```

Comment:
Note very few of our _new_ middleware take in options directly. They only do this if they expect to have more than one middleware instance in the pipeline (static files). Most handle options via DI.

Reply:
Yeah the pattern of taking in an instance tends to be somewhat confusing since the normal Configure ProblemDetailsOptions in DI will have no affect on this instance

Reply:
> Note very few of our new middleware take in options directly. They only do this if they expect to have more than one middleware instance in the pipeline (static files). Most handle options via DI.

Got it. I wasn't aware of this. I'll adjust accordingly.